src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -197,40 +254,78 @@ struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
+    /// the maximum allowed
+    /// @param deleted_scenes [out] bool letting the caller (in this case the load method) know wether or not it was necessary to
+    /// delete scenes from the memory
+    /// @return CHIP_NO_ERROR on success, specific CHIP_ERROR otherwise
+    CHIP_ERROR DeserializeAdjust(TLV::TLVReader & reader, PersistentStorageDelegate * storage, bool & deleted_scenes)
     {
         ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
-
         TLV::TLVType fabricSceneContainer;
         ReturnErrorOnFailure(reader.EnterContainer(fabricSceneContainer));
-
         ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kSceneCount)));
         ReturnErrorOnFailure(reader.Get(scene_count));
+        scene_count = min(scene_count, max_scenes_per_fabric);
         ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::ContextTag(TagScene::kStorageIDArray)));
         TLV::TLVType sceneMapContainer;
         ReturnErrorOnFailure(reader.EnterContainer(sceneMapContainer));
 
         uint8_t i = 0;
         CHIP_ERROR err;
-        while ((err = reader.Next(TLV::AnonymousTag())) == CHIP_NO_ERROR && i < kMaxScenesPerFabric)
+        uint8_t deleted_scenes_count = 0;
+        while ((err = reader.Next(TLV::AnonymousTag())) == CHIP_NO_ERROR)
         {
             TLV::TLVType sceneIdContainer;
-            ReturnErrorOnFailure(reader.EnterContainer(sceneIdContainer));
-            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kEndpointID)));
-            ReturnErrorOnFailure(reader.Get(scene_map[i].mEndpointId));
-            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kGroupID)));
-            ReturnErrorOnFailure(reader.Get(scene_map[i].mGroupId));
-            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kSceneID)));
-            ReturnErrorOnFailure(reader.Get(scene_map[i].mSceneId));
-            ReturnErrorOnFailure(reader.ExitContainer(sceneIdContainer));
+            if (i < max_scenes_per_fabric)
+            {
+                ReturnErrorOnFailure(reader.EnterContainer(sceneIdContainer));
+                ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kEndpointID)));
+                ReturnErrorOnFailure(reader.Get(scene_map[i].mEndpointId));
+                ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kGroupID)));
+                ReturnErrorOnFailure(reader.Get(scene_map[i].mGroupId));
+                ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kSceneID)));
+                ReturnErrorOnFailure(reader.Get(scene_map[i].mSceneId));
+                ReturnErrorOnFailure(reader.ExitContainer(sceneIdContainer));
+            }
+            else
+            {
+                SceneTableData scene(fabric_index, i);
+                ReturnErrorOnFailure(reader.EnterContainer(sceneIdContainer));
```

Comment:
So once we have deleted one scene, if something here fails we will return and:

1. Not do the `GlobalSceneCount`.
2. Not set `deleted_scenes` to true.

right?  Is that ok?  Seems like it might get us into having inconsistent storage states....

Reply:
Moved the GlobalSceneCount adjustment outside of the Deserialize function into load. That way it will still get called if scenes were deleted before returning.